tests/Middleware/HttpCacheTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Middleware;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Nyholm\Psr7\Response;
+use Nyholm\Psr7\ServerRequest;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Http\Method;
+use Yiisoft\Yii\Web\Middleware\HttpCache;
+
+class HttpCacheTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function validCacheResult(): void
+    {
+        $time = \time();
+        $middleware = $this->createMiddlewareWithLastModified($time);
+        $headers = [
+            'If-Modified-Since' => $time,
+        ];
+        $response = $middleware->process($this->createServerRequest(Method::GET, $headers), $this->createRequestHandler());
+        $this->assertEquals(200, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function invalidCacheResult(): void
```

Comment:
304 is not invalid cache. It's, in fact, valid i.e. result was not modified.

In this case:

1. Body should be empty.
2. Last-Modified header should be present.